include/libzfs.h

```diff
@@ -732,7 +732,7 @@ extern int zfs_unshare_nfs(zfs_handle_t *, const char *);
 extern int zfs_unshare_smb(zfs_handle_t *, const char *);
 extern int zfs_unshareall_nfs(zfs_handle_t *);
 extern int zfs_unshareall_smb(zfs_handle_t *);
-extern int zfs_unshareall_bypath(zfs_handle_t *, const char *);
+extern int zfs_unshareall_bypath(zfs_handle_t *, const char *, const char *);
```

Comment:
Let's just add a new function `zfs_unshareall_bytype()` and leave `zfs_unshareall_bypath()` as is.